src/core_atmosphere/physics/mpas_atmphys_control.F

```diff
@@ -493,6 +495,79 @@ subroutine physics_compatibility_check(dminfo, blockList, streamManager, ierr)
 
  end subroutine physics_compatibility_check
 
+!=================================================================================================================
+!  routine sst_update_check()
+!
+!> \brief Sea Surface Interval Update Check
+!> \author Miles Curry
+!> \date   January 2019
+!> \details
+!>  This routine checks that the user has correctly set the SST update options,
+!>  by setting the `config_sst_update` namelist option equal to `true` .AND.
+!>  by setting the `input_interal` surface stream variable in streams.atmosphere
+!>  to a valid HH:MM:SS update time.
+!>
+ subroutine sst_update_check(blockList, streamManager, ierr)
+!=================================================================================================================
+
+ use mpas_timekeeping, only : mpas_split_string
+
+ implicit none
+
+ ! Input Variables
+ type (block_type), pointer :: blockList
+ type (MPAS_streamManager_type), pointer :: streamManager
+ integer, intent(out) :: ierr
+
+ ! Local Variables
+ logical,pointer:: config_sst_update
+ character(len=StrKIND) :: stream_interval
+
+
+ call mpas_pool_get_config(blocklist % configs, 'config_sst_update', config_sst_update)
+ call MPAS_stream_mgr_get_property(streamManager, 'surface', MPAS_STREAM_PROPERTY_RECORD_INTV, stream_interval, &
+                                   direction=MPAS_STREAM_INPUT, ierr=ierr)
+
+ write(0,*) "stream_interval is: ", stream_interval
+ write(0,*) "config_sst_update is: ", config_sst_update
+
+
+ if ( config_sst_update .AND. stream_interval == 'none' ) then
+    call mpas_log_write('*******************************************************************************', &
+                        messageType=MPAS_LOG_ERR)
+    call mpas_log_write("'config_sst_update' in the physics section of the namelist.atmosphere file was set", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("to 'true', but the 'input_interval' in the surface stream of the streams.atmosphere", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("file has not been set.", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("Please set the update interval to a valid DD_HH:MM:SS interval or turn 'config_sst_update'", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("to 'false'.", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write('*******************************************************************************', &
+                        messageType=MPAS_LOG_ERR)
+    call mpas_log_write("FATAL ERROR: SST Interval Update Error - see above", &
+                        messageType=MPAS_LOG_CRIT)
```

Comment:
I think we may actually want to omit the critical error message here so that the error can bubble back up the call stack, possibly letting other errors be reported before the model stops. A non-zero return code from this routine will cause a non-zero return code in `mpas_atm_run_compatibility`, which will cause a non-zero return code in `atm_core_init`, which will finally stop the model in `mpas_init` (in `mpas_subdriver.F`).